nil pointer dereference when misconfigured storage used

nxs-backup 3.8.2

Given following `/etc/nxs-backup/nxs-backup.conf`:
```yaml
server_name: PristineAbobus

notifications:
  mail:
    enabled: false
  webhooks: []

storage_connects:
  - name: pivo
    scp_params:
      user: backup
      host: 10.11.12.13
      port: 22
      password: my_ssh_password
      key_file: "not_existing_file"

jobs:
  - job_name: some-files
    type: desc_files

    sources:
      - name: desc_files
        targets: [ /etc/nxs-backup ]
        excludes: []

    storages_options:
      - storage_name: pivo
        backup_path: backups/
        retention:
          days: 7
          weeks: 5
          months: 5

logfile: stdout
```
Running `nxs-backup start` outputs:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xeb9d04]

goroutine 1 [running]:
github.com/nixys/nxs-backup/modules/storage/sftp.(*SFTP).Clone(0x0)
        .../nxs-backup/modules/storage/sftp/sftp.go:432 +0x24
github.com/nixys/nxs-backup/ctx.jobsInit({0xc00042a9b0, 0xc000050ee0, {0xc00017e870, 0x1, 0x1}, 0xc00067a810})
        .../nxs-backup/ctx/jobs_init.go:102 +0x952
github.com/nixys/nxs-backup/ctx.appInit(0xc00042a2d0, {0x1002293, 0x1f})
        .../nxs-backup/ctx/context.go:223 +0x694
github.com/nixys/nxs-backup/ctx.AppCtxInit()
        .../nxs-backup/ctx/context.go:110 +0x293
github.com/nixys/nxs-go-appctx/v3.(*AppCtx).Run(0xc00048de30)
        /root/go/pkg/mod/github.com/nixys/nxs-go-appctx/v3@v3.0.0/appctx.go:65 +0x36
main.main()
        .../nxs-backup/main.go:38 +0x4d6
```

--- a/ctx/storages_init.go
+++ b/ctx/storages_init.go
@@ -28,8 +28,9 @@ var allowedConnectParams = []string{
 
 func storagesInit(storageConnects []storageConnectConf, mainLim *limitsConf) (storagesMap map[string]interfaces.Storage, err error) {
 	var (
-		rl   int64
-		errs *multierror.Error
+		rl      int64
+		errs    *multierror.Error
+		storage interfaces.Storage
 	)
 
 	storagesMap = make(map[string]interfaces.Storage)
@@ -57,42 +58,27 @@ func storagesInit(storageConnects []storageConnectConf, mainLim *limitsConf) (st
 
 		switch {
 		case st.S3Params != nil:
-			storagesMap[st.Name], err = s3.Init(st.Name, s3.Opts(*st.S3Params), rl)
-			if err != nil {
-				errs = multierror.Append(errs, fmt.Errorf("Failed to init storage `%s` with error: %w ", st.Name, err))
-			}
+			storage, err = s3.Init(st.Name, s3.Opts(*st.S3Params), rl)
 		case st.ScpParams != nil:
-			storagesMap[st.Name], err = sftp.Init(st.Name, sftp.Opts(*st.ScpParams), rl)
-			if err != nil {
-				errs = multierror.Append(errs, fmt.Errorf("Failed to init storage `%s` with error: %w ", st.Name, err))
-			}
+			storage, err = sftp.Init(st.Name, sftp.Opts(*st.ScpParams), rl)
 		case st.SftpParams != nil:
-			storagesMap[st.Name], err = sftp.Init(st.Name, sftp.Opts(*st.SftpParams), rl)
-			if err != nil {
-				errs = multierror.Append(errs, fmt.Errorf("Failed to init storage `%s` with error: %w ", st.Name, err))
-			}
+			storage, err = sftp.Init(st.Name, sftp.Opts(*st.SftpParams), rl)
 		case st.FtpParams != nil:
-			storagesMap[st.Name], err = ftp.Init(st.Name, ftp.Opts(*st.FtpParams), rl)
-			if err != nil {
-				errs = multierror.Append(errs, fmt.Errorf("Failed to init storage `%s` with error: %w ", st.Name, err))
-			}
+			storage, err = ftp.Init(st.Name, ftp.Opts(*st.FtpParams), rl)
 		case st.NfsParams != nil:
-			storagesMap[st.Name], err = nfs.Init(st.Name, nfs.Opts(*st.NfsParams), rl)
-			if err != nil {
-				errs = multierror.Append(errs, fmt.Errorf("Failed to init storage `%s` with error: %w ", st.Name, err))
-			}
+			storage, err = nfs.Init(st.Name, nfs.Opts(*st.NfsParams), rl)
 		case st.WebDavParams != nil:
-			storagesMap[st.Name], err = webdav.Init(st.Name, webdav.Opts(*st.WebDavParams), rl)
-			if err != nil {
-				errs = multierror.Append(errs, fmt.Errorf("Failed to init storage `%s` with error: %w ", st.Name, err))
-			}
+			storage, err = webdav.Init(st.Name, webdav.Opts(*st.WebDavParams), rl)
 		case st.SmbParams != nil:
-			storagesMap[st.Name], err = smb.Init(st.Name, smb.Opts(*st.SmbParams), rl)
-			if err != nil {
-				errs = multierror.Append(errs, fmt.Errorf("Failed to init storage `%s` with error: %w ", st.Name, err))
-			}
+			storage, err = smb.Init(st.Name, smb.Opts(*st.SmbParams), rl)
 		default:
-			errs = multierror.Append(errs, fmt.Errorf("unable to define `%s` storage connect type by its params. Allowed connect params: %s", st.Name, strings.Join(allowedConnectParams, ", ")))
+			err = fmt.Errorf("unable to define `%s` storage connect type by its params. Allowed connect params: %s", st.Name, strings.Join(allowedConnectParams, ", "))
+		}
+
+		if err != nil {
+			errs = multierror.Append(errs, fmt.Errorf("Failed to init storage `%s` with error: %w ", st.Name, err))
+		} else {
+			storagesMap[st.Name] = storage
 		}
 	}
 

